alerts/machine-config-operator/MachineConfigControllerDrainError.md

```diff
@@ -0,0 +1,114 @@
+# MCCDrainError
+
+## Meaning 
+
+Alerts the user to when the Machine Config Operator 
+failed to drain a node. Preventing the MCO from restarting 
+the node. 
+
+>Alerts the user to a failed node drain. Always triggers when the failure
+>happens one or more times.
+
+This alert will fire as a warning when: 
+
+- If the mco is unable to drain the node for an hour
+- When the failure occurs one or more times in a node 
+
+## Impact 
+
+If the mco fails to drain a node it will be unable to reboot the node. 
+Preventing any changes via MachineConfig or a cluster upgrade to be applied. 
+This will prevent a cluster upgrade from progressing/completing.
+
+## Diagnosis 
+
+If a node is failing to draining the first place the alert itself recommends to check the machine-config-controller pod inside the openshift-machine-config-operator namespace. 
+The machine-config-controller pod is the central point of management for incoming updates to 
+machines.
+
+For the commands below replace the $CONTROLLERPOD variable with the name of your own machine-config-controller pod name. 
+
+```console
+oc -n openshift-machine-config-operator logs $CONTROLLERPOD -c machine-config-controller
+```
+
+When a drain is initiated the following log will be recorded.
+
+```console
+1 drain_controller.go:173] node xxxxxxx-xxxxxx-xxxxxx-xxxxx: initiating drain
+```
+The controller will then log each and every pod drained from the node. 
+
+```console
+  1 drain_controller.go:173] node xxxxxxx-xxxxxx-xxxxxx-xxxxx: Evicted pod <namespace>/<pod-name>
+```
+
+If the controller is unable to drain a pod after 1m30s then it will fire an error log.
```

Comment:
Does the MCC do the drain?
```suggestion
If the MCO/MCC is unable to drain a pod after 1m30s, it logs the following error message:
```

Reply:
As far as I know yes, the machine config controller is what initiates and does drains for nodes but we can just specify MCO/MCC as a catch all